### DISCUSSION TITLE
Higraphs as a new model type

### DISCUSSION THREAD

**Discussion by ghillebrand** [gaphor/gaphor#2814] [Ideas]
Is there anyone working on/ interested in making Gaphor able to model generic [Higraphs](https://dl.acm.org/doi/pdf/10.1145/42411.42414)? 
Higraphs were more general precursors to what became UML State Diagrams (David Harel worked on both, I think).
It seems that this could be implemented as a modelling language as discussed [here](https://github.com/gaphor/gaphor/issues/1235), which refers to [https://gaphor.readthedocs.io/en/latest/modeling_language.html](https://gaphor.readthedocs.io/en/latest/modeling_language.html).

I am currently bending UML State Diagrams to do this, but ideally one needs a few more abilities, some which are modelling semantics (eg more general set intersections, vertical blob orthogonality), and some which would require a Gaphas extension (spline edges with multiple starts and ends - "hyperedges" in Harel's terms). Some of the algebraic underpinning could likely be done as plugins.

I am happy to start tinkering slowly to understand Gaphor better to work on this  - most probably with some reporting plugins on State diagrams. 
Higraphs are exceptionally powerful tools, and sadly neglected in the general modelling space.

**Comment by danyeaw** [comment]
Hi @ghillebrand, thanks for the idea! If you would like to help implement this, I think we would be open to adding another diagram type.
